### ISSUE TITLE
90th, 99th and 90.5th latency percentile are not showing always

### ISSUE THREAD

**Issue by JagmohanSharma** [Netflix/Hystrix#1525]
In our Hystrix dashboard , these values are not being shown always as for a fraction of time we see certain values but on next second it gets value as 0 again which it keeps for some time.It seems to be intermittent, Is this an expected behaviour OR we should look into this to find the cause or validness?

**Comment by mattrjacobs** [comment]
The dashboard consumes a metric stream from your application running Hystrix commands.  The stream emits a steady flow of metrics.  If no commands occur in a given time period, then all latency values are sent as 0.  Is it possible this is the case you're in?

**Comment by JagmohanSharma** [comment]
Do you mean if there is no request is being served by hystrix command in a given period of time then latency value sent as 0?   
![githubattachment](https://cloud.githubusercontent.com/assets/19706178/24817311/a9f34f2e-1bf9-11e7-9b4b-f631926f4a61.jpg)

But as per screen shot attached , the cluster serving lot of request shows latency as 0 for most of the time but the cluster having 0 traffic showing high number latency?

is this expected?

**Comment by mukteshkrmishra** [comment]
There are 2 types of fields showing in dashboard. One which are related with calls and second is aggregated fields. Values associated with calls reset to 0 as soon as we stop making calls via hystrix. Aggregate values are always shown no matter u stoppd making calls from hystrix or not. (Same behavior in metrics stream also).

**Comment by JagmohanSharma** [comment]
@mukteshkrmishra  Actually for one cluster which is receiving requests, latency is 0 most of the times which we are expecting to be at some level of ms value. 
You might be true for some fields but that cant be the case with latency as this can be of one type of field only. so which type of field is latency as per your comment- related to calls Or aggregated?

**Comment by mukteshkrmishra** [comment]
We have 2 flavors of method: rolling metrics and cumulative metrics. Rolling metrics track counters/latency over a rolling window of time and cumulative metrics track counters since the JVM start.  For cumulative values, those never lose data, so I would expect them to stay the same even with no traffic. For rolling count ("rollingCountThreadsExecuted, rollingCountCommandsRejected etc)  you should see values as 0 as soon as you stop making calls.

Moreover, the window for rolling is a bounded amount of time (60s is the default). The window for cumulative is infinite - it starts when the JVM starts and goes until the JVM stops. Therefore, traffic leaving will not increase cumulative metrics anymore, just leave them at their current values. Cumulative counters cannot go down.  

Does this answer your question @JagmohanSharma 

cc: @mattrjacobs

**Comment by mattrjacobs** [comment]
All of the values that the dashboard consumes are rolling.  So if you were to stop traffic to an instance, I would expect that instance to have 0s for all metrics.

Is it possible that the commands you're measuring actually have a 0ms latency?

**Comment by JagmohanSharma** [comment]
@mattrjacobs Actually we are expecting certain ms of latency for our command executions. But most of of time even in a high traffic volume, we are getting latency value as 0 ms. and very few times it shows some value for latency. 

Please correct me if I am wrong here. The screenshot which I have attached in my previous comment contains turbine stream data, so this is aggregating metrics data for all service instances. Now in case of hystrix stream for any specific service instance, in case of traffic is stopped then we can expect that to have 0ms as latency. But in case of turbine dashboard, as shown in image for first command , aggregated traffic volume is 1415 so this must have certain value for latency most of the time.

**Comment by mattrjacobs** [comment]
Yeah, it certainly looks like a bug, though one I've not seen before.  Can you try running an instance of the dashboard against a single instance's metrics stream?  If that works and the aggregated version does not, that would imply a bug in the aggregation.  If pointing it to a single instance does not work, then I can investigate where the metrics are getting dropped.

**Comment by JagmohanSharma** [comment]
@mattrjacobs I checked it with pointing to single instance, most of the time this does not show any value for latency percentile. So aggregated version also does not show any value for this. It would be worth to check this for single instance as aggregation seems to work properly. Please let me know if you require more information.

**Comment by mattrjacobs** [comment]
Based on your description, I'm stumped.  Can you post a full dump of the stream data you're receiving?
